packages/mui-system/src/styleFunctionSx/defaultSxConfig.js

```diff
@@ -0,0 +1,234 @@
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import { getValue } from '../style';
+import { handleBreakpoints } from '../breakpoints';
+import { paddingKeys, padding, marginKeys, margin } from '../spacing';
+import { borderRadius, borderTransform } from '../borders';
+import { gap, rowGap, columnGap } from '../cssGrid';
+import { paletteTransform } from '../palette';
+import { maxWidth, sizingTransform } from '../sizing';
+
+const defaultSxConfig = {
+  // borders
+  border: {
+    themeKey: 'borders',
+    transform: borderTransform,
+  },
+
+  borderTop: {
+    themeKey: 'borders',
+    transform: borderTransform,
+  },
+
+  borderRight: {
+    themeKey: 'borders',
+    transform: borderTransform,
+  },
+
+  borderBottom: {
+    themeKey: 'borders',
+    transform: borderTransform,
+  },
+
+  borderLeft: {
+    themeKey: 'borders',
+    transform: borderTransform,
+  },
+
+  borderColor: {
+    themeKey: 'palette',
+  },
+
+  borderTopColor: {
+    themeKey: 'palette',
+  },
+
+  borderRightColor: {
+    themeKey: 'palette',
+  },
+
+  borderBottomColor: {
+    themeKey: 'palette',
+  },
+
+  borderLeftColor: {
+    themeKey: 'palette',
+  },
+
+  borderRadius: {
+    themeKey: 'shape.borderRadius',
+    style: borderRadius,
+  },
+
+  // palette
+  color: {
+    themeKey: 'palette',
+    transform: paletteTransform,
+  },
+  bgcolor: {
+    themeKey: 'palette',
+    cssProperty: 'backgroundColor',
+    transform: paletteTransform,
+  },
+  backgroundColor: {
+    themeKey: 'palette',
+    transform: paletteTransform,
+  },
+
+  // spacing
+  ...paddingKeys.reduce((styleDefinitions, paddingKey) => {
+    return {
+      ...styleDefinitions,
+      [paddingKey]: {
+        style: padding,
+      },
+    };
+  }, {}),
+
+  ...marginKeys.reduce((styleDefinitions, marginKey) => {
+    return {
+      ...styleDefinitions,
+      [marginKey]: {
+        style: margin,
+      },
+    };
+  }, {}),
+
+  // display
+  displayPrint: {
+    cssProperty: false,
+    transform: (value) => ({
+      '@media print': {
+        display: value,
+      },
+    }),
+  },
+  display: {},
+  overflow: {},
+  textOverflow: {},
+  visibility: {},
+  whiteSpace: {},
+
+  // flexbox
+  flexBasis: {},
+  flexDirection: {},
+  flexWrap: {},
+  justifyContent: {},
+  alignItems: {},
+  alignContent: {},
+  order: {},
+  flex: {},
+  flexGrow: {},
+  flexShrink: {},
+  alignSelf: {},
+  justifyItems: {},
+  justifySelf: {},
+
+  // grid
+  gap: {
+    style: gap,
+  },
+  rowGap: {
+    style: rowGap,
+  },
+  columnGap: {
+    style: columnGap,
+  },
+  gridColumn: {},
+  gridRow: {},
+  gridAutoFlow: {},
```

Comment:
Is there a way we can drop all these keys? I wonder if we can't reduce the bundle size footprint.

Reply:
Also wonder what's the `{}` for?

Reply:
To allow breakpoints syntax.